client/app/bundles/course/survey/reducers/surveyForm.js

```diff
@@ -18,11 +18,14 @@ export default function (state = initialState, action) {
     case actionTypes.SURVEY_FORM_HIDE: {
       return { ...state, visible: false };
     }
-    case actionTypes.CREATE_SURVEY_REQUEST: {
+    case actionTypes.CREATE_SURVEY_REQUEST:
+    case actionTypes.UPDATE_SURVEY_REQUEST: {
       return { ...state, disabled: true };
     }
     case actionTypes.CREATE_SURVEY_SUCCESS:
-    case actionTypes.CREATE_SURVEY_FAILURE: {
+    case actionTypes.CREATE_SURVEY_FAILURE:
```

Comment:
Haha this was the source of the issue last time right, where the surveys can fail silently. I haven't had the time to go and fix it yet - were you able to fix it along the way?

Reply:
Ah there is no behavior change here.. I am just adding reducer actions for survey update request/success/failure.